Tasks/NuGetCommandV2/nugetpublisher.ts

```diff
@@ -439,17 +439,25 @@ function getAccessToken(isInternalFeed: boolean): string{
         {           
             tl.debug("Checking for auth from Cred Provider."); 
             const feed = getProjectAndFeedIdFromInputParam('feedPublish');
+            const feedUrl: string = await nutil.getNuGetFeedRegistryUrl(
+                packagingLocation.DefaultPackagingUri,
+                feed.feedId,
+                feed.projectId,
+                null /* default to V3 */,
+                null /* no accessToken  */,
+                false /* useSession */);
+
             const JsonEndpointsString = process.env["VSS_NUGET_EXTERNAL_FEED_ENDPOINTS"];
+
             if (JsonEndpointsString) {
+                tl.debug(`Feed details ${feed.feedId} ${feed.projectId}`);
                 tl.debug(`Endpoints found: ${JsonEndpointsString}`);
-
                 let endpointsArray: { endpointCredentials: EndpointCredentials[] } = JSON.parse(JsonEndpointsString);
-                tl.debug(`Feed details ${feed.feedId} ${feed.projectId}`);
 
-                for (let endpoint_in = 0; endpoint_in < endpointsArray.endpointCredentials.length; endpoint_in++) {
-                    if (endpointsArray.endpointCredentials[endpoint_in].endpoint.search(feed.feedName) != -1) {
-                        tl.debug(`Endpoint Credentials found for ${feed.feedName}`);
-                        accessToken = endpointsArray.endpointCredentials[endpoint_in].password;
+                for (const e of endpointsArray.endpointCredentials) {
+                    if (e.endpoint == feedUrl) {
```

Comment:
```suggestion
                    if (e.endpoint === feedUrl) {
```

That said, I wouldn't consider the feed URL to be stable enough for the purpose of matching -- there are lots of ways to vary a feed URL and still refer to the same feed (names vs ids, visualstudio.com vs dev.azure.com, optional DefaultCollection, casing, all 4 combinations of guids vs names)

Reply:
Hm are you saying there is potential for variations in the feed URL being created in line 442, or there is possible variation in the service connection being retrieved?

Reply:
both, really. When asking the Location Service to form a URL, there can be variations in what URL is formed depending on the caller's context.

Reply:
I'm now cross-checking feed-service connection mappings by making a call to the service index and verifying a match in feed/project. Guidance will be given to customer to use projectName/feedName when targeting the feed within a task. This keeps us from handling the case where GUIDs are used.